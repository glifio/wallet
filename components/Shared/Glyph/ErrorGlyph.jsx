import React, { forwardRef } from 'react'
import Glyph from '.'

const ErrorGlyph = forwardRef(({ ...props }, ref) => (
  <Glyph
    {...props}
    ref={ref}
    acronym='Er'
    backgroundColor='status.fail.background'
    color='status.fail.foreground'
  />
))

export default ErrorGlyph
