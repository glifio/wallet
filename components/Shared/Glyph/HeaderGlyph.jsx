import React from 'react'
import { string } from 'prop-types'
import Box from '../Box'
import { IconGlif } from '../Icons'
import { Header } from '../Typography'

const HeaderGlyph = ({ alt, color, text, imageUrl, fill, imageOpacity }) => {
  return (
    <Box
      position='relative'
      display='flex'
      alignItems='center'
      justifyContent='space-between'
      py={2}
      mr={[0, 6]}
      mb={[3, 0]}
      px={3}
      borderRadius={4}
      width={['100%', 'auto']}
      height='120px'
      opacity='1'
      css={`
        &:before {
          content: '';
          position: absolute;
          top: 0px;
          right: 0px;
          bottom: 0px;
          left: 0px;
          background-color: rgba(0, 0, 0, 0.25);
          background: url(${imageUrl}) center no-repeat;
          background-size: 100%;
          border-radius: 16px;
          opacity: ${imageOpacity};
          z-index: -9;
          alt: ${alt};
        }
      `}
    >
      <IconGlif
        fill={fill}
        size={7}
        css={`
          transform: rotate(-90deg);
        `}
      />
      <Header color={color} ml={3}>
        {text}
      </Header>
    </Box>
  )
}

HeaderGlyph.propTypes = {
  alt: string,
  text: string.isRequired,
  imageUrl: string.isRequired,
  imageOpacity: string,
  color: string,
  fill: string
}

HeaderGlyph.defaultProps = {
  alt: '',
  imageOpacity: '1',
  color: 'core.black',
  fill: '#000'
}

export default HeaderGlyph
