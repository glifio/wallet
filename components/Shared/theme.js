export const baseColors = {
  mono: {
    black: '#000',
    nearblack: '#262626',
    darkgray: '#666666',
    lightgray: '#C4C4C4',
    silver: '#999999',
    white: '#ffffff',
    transparent: 'transparent'
  },
  blue: {
    lightest: '#F6F8FE',
    lighter: '#EFF3FD',
    light: '#E4EBFC',
    mid: '#d1ddfa',
    muted: '#e8ecf7'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    dark: '#007056',
    darker: '#08442F'
  },
  red: {
    light: '#FC6D6F',
    dark: '#660000'
  },
  orange: {
    light: '#FF9E80',
    dark: '#661800'
  },
  yellow: {
    light: '#FFDC99',
    deep: '#FCA703'
  },
  purple: {
    light: '#E0D7FE',
    deep: '#5E26FF'
  }
}

// The core color object is the only non-semantic object here. This is because these colors are used so widely that it would be highly redundant to replicate these color values repeatedly within this "colors" object to style the text/background of every single Component the app renders.
const core = {
  primary: baseColors.purple.deep,
  secondary: baseColors.purple.light,
  black: baseColors.mono.black,
  nearblack: baseColors.mono.nearblack,
  darkgray: baseColors.mono.darkgray,
  lightgray: baseColors.mono.lightgray,
  silver: baseColors.mono.silver,
  white: baseColors.mono.white,
  transparent: baseColors.mono.transparent
}

const colors = {
  core,
  buttons: {
    primary: {
      background: baseColors.green.primary,
      borderColor: baseColors.green.primary,
      color: baseColors.green.darker
    },
    secondary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.mono.nearblack,
      color: baseColors.mono.nearblack
    },
    tertiary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.purple.light,
      color: baseColors.purple.light
    },
    'tertiary-selected': {
      background: baseColors.purple.primary,
      borderColor: baseColors.purple.primary,
      color: baseColors.purple.light
    }
  },
  background: {
    app: baseColors.blue.lightest,
    screen: baseColors.blue.lighter,
    messageHistory: baseColors.blue.muted
  },
  card: {
    account: {
      background: core.primary,
      color: core.secondary
    },
    balance: {
      background: baseColors.mono.transparent,
      color: core.nearblack
    },
    confirmation: {
      background: baseColors.green.light,
      foreground: baseColors.green.dark
    },
    error: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    }
  },
  input: {
    background: {
      base: baseColors.blue.light,
      active: baseColors.blue.mid,
      valid: baseColors.green.light,
      invalid: baseColors.red.light
    },
    border: core.silver
  },
  status: {
    success: {
      background: baseColors.green.primary,
      foreground: baseColors.green.darker
    },
    pending: {
      background: baseColors.yellow.deep,
      foreground: baseColors.mono.darkgray
    },
    warning: {
      background: baseColors.orange.light,
      foreground: baseColors.orange.dark
    },
    fail: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    },
    inactive: baseColors.mono.lightgray
  }
}

const theme = {
  colors,
  fontSizes: ['0rem', '1rem', '1.125rem', '1.25rem', '1.5rem', '2rem', '3rem'],
  fontWeights: [0, 400, 600, 900],
  letterSpacings: [0, 1, 2, 4, 8],
  lineHeights: {
    solid: 1,
    title: 1.2,
    copy: 1.4
  },
  textStyles: {
    bigTitle: {
      fontSize: 6,
      fontWeight: 700,
      margin: 0,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Medium'
    },
    title: {
      fontSize: 4,
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    text: {
      fontSize: 3,
      fontColor: core.nearblack,
      fontWeight: 400,
      lineHeight: 'copy',
      fontFamily: 'RT-Alias-Grotesk'
    },
    label: {
      fontSize: 2,
      fontColor: core.darkgray,
      fontWeight: 400,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Grotesk'
    },
    num: {
      l: {
        fontSize: 5,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Medium'
      },
      xl: {
        fontSize: 6,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Medium'
      }
    }
  },
  fonts: {
    AliasMedium:
      '"RT-Alias-Medium", "system-ui", "Segoe UI", "Roboto", Helvetica',
    AliasGrotesk:
      '"RT-Alias-Grotesk", "system-ui", "Segoe UI", "Roboto", Helvetica',
    sansSerif:
      '"system-ui", "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";',
    mono:
      '"system-ui", "Segoe UI", Roboto Mono, Helvetica, Arial, monospace, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";'
  },
  space: [0, 4, 8, 16, 24, 32, 48, 64, 128, 256],
  sizes: [
    0,
    4,
    8,
    16,
    24,
    32,
    48,
    64,
    80,
    160,
    240,
    320,
    480,
    560,
    640,
    720,
    800,
    960,
    1024,
    1200
  ],
  radii: ['0', '1px', '4px', '8px', '16px', '32px', '100px'],
  borders: [0, `1px solid`, `4px solid`],
  borderStyles: ['solid'],
  borderWidths: ['0', '1px', '2px', '4px'],
  breakpoints: ['40em', '52em', '64em'],
  shadows: [
    '0',
    '0 0.7px 2.2px -8px rgba(0, 0, 0, 0.18),0 1.7px 5.4px  rgba(0, 0, 0, 0.129),0 3.1px 10.1px rgba(0, 0, 0, 0.107),0 5.6px 18.1px rgba(0, 0, 0, 0.09),0 10.4px 33.8px rgba(0, 0, 0, 0.073),0 25px 81px rgba(0, 0, 0, 0.051)',
    'rgba(0, 0, 0, 0.10) 0px 0.7px 2.2px -8px, rgba(0, 0, 0, 0.04) 0px 1.7px 2.4px, rgba(0, 0, 0, 0.106) 0px 3.1px 8.1px, rgba(0, 0, 0, 0.04) 0px 5.6px 12.1px, rgba(0, 0, 0, 0.045) 0px 4.4px 4.8px, rgba(0, 0, 0, 0.05) 0px 15px 41px'
  ],
  opacity: {
    disabled: 0.4
  },
  zIndices: [0, 9, 99, 999, 9999]
}

export default theme
