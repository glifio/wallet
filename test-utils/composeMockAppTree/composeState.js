import cloneDeep from 'lodash.clonedeep'
import { FilecoinNumber } from '@openworklabs/filecoin-number'

import createPath from '../../utils/createPath'
import { IMPORT_MNEMONIC } from '../../constants'
import { initialState } from '../../store/states'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'

export const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  postOnboardWithError: cloneDeep({
    ...initialState,
    error: 'error for testing',
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  })
}

export const composeWalletProviderState = (
  initialWalletProviderState,
  preset
) => {
  switch (preset) {
    case 'postOnboard': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    case 'postOnboardWithError': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    default:
      return initialWalletProviderState
  }
}
