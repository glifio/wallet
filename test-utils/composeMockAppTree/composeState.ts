import cloneDeep from 'lodash.clonedeep'
import { FilecoinNumber } from '@glif/filecoin-number'

import createPath from '../../utils/createPath'
import { IMPORT_MNEMONIC } from '../../constants'
import { initialState } from '../../store/states'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'
import { emptyMsigState } from '../../MsigProvider/types'
import { WALLET_ADDRESS, MULTISIG_ACTOR_ADDRESS, signers } from '../constants'

export const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  postOnboardLowBal: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('.000001', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  postOnboardWithError: cloneDeep({
    ...initialState,
    error: 'error for testing',
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  pendingMsigCreate: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0,
    messages: {
      confirmed: [],
      pending: [
        {
          cid: 'bafy2bzaced2godar6dht6ag6omfa53jr6tykygtzskzodqo2k4ngvyou4yct2',
          from: 'f1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a',
          gasFeeCap: '100425',
          gasLimit: 11452096,
          gasPremium: '99371',
          maxFee: '1150076740800',
          method: 'EXEC',
          nonce: 56,
          paidFee: '0',
          params: {
            num_approvals_threshold: 1,
            signers: ['f1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a'],
            start_epoch: '0',
            unlock_duration: '0'
          },
          timestamp: 1632317830,
          to: 'f01'
        }
      ]
    }
  })
}

export const composeWalletProviderState = initialWalletProviderState => {
  return Object.freeze({
    ...initialWalletProviderState,
    walletType: IMPORT_MNEMONIC,
    walletProvider: mockWalletProviderInstance
  })
}

export const mockMsigProviderContext = {
  Address: MULTISIG_ACTOR_ADDRESS,
  ActorCode: 'fil/5/multisig',
  Balance: new FilecoinNumber('1', 'fil'),
  AvailableBalance: new FilecoinNumber('1', 'fil'),
  Signers: signers,
  InitialBalance: new FilecoinNumber('1', 'fil'),
  NextTxnID: 0,
  NumApprovalsThreshold: 1,
  StartEpoch: 0,
  UnlockDuration: 0,
  errors: {
    notMsigActor: false,
    connectedWalletNotMsigSigner: false,
    actorNotFound: false,
    unhandledError: ''
  },
  loading: false,
  setMsigActor: null
}

export const composeMsigProviderState = (preset: keyof typeof presets) => {
  switch (preset) {
    case 'preOnboard': {
      return Object.freeze(emptyMsigState)
    }
    case 'pendingMsigCreate': {
      return Object.freeze(emptyMsigState)
    }
    default: {
      return Object.freeze(mockMsigProviderContext)
    }
  }
}
