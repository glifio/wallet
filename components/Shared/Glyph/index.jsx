import React, { forwardRef } from 'react'
import styled from 'styled-components'
import { string, object } from 'prop-types'
import { baseColors } from '../theme'
import Box from '../Box'

const GlyphText = styled.h3`
  font-family: ${props => props.theme.fonts.AliasMedium};
  font-weight: 700;
  font-size: ${props => props.theme.fontSizes[4]};
`

const Glyph = forwardRef(
  ({ acronym, Icon, color, backgroundColor, ...props }, ref) => (
    <Box
      display='flex'
      alignItems='center'
      justifyContent='center'
      size={6}
      borderWidth='0.1875rem'
      borderRadius={3}
      borderStyle={0}
      backgroundColor={backgroundColor}
      color={color}
      ref={ref}
      {...props}
    >
      {acronym ? <GlyphText>{acronym}</GlyphText> : <Icon />}
    </Box>
  )
)

Glyph.propTypes = {
  /**
   * The two letters displayed in the glyph
   */
  acronym: string,
  /**
   * This is a dom element, but PropTypes.node was failing
   */
  Icon: object,
  /**
   * The color of the text and border
   */
  color: string,
  backgroundColor: string
}

Glyph.defaultProps = {
  color: baseColors.mono.black,
  acronym: ''
}

export default Glyph
