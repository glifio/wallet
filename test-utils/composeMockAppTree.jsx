/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import cloneDeep from 'lodash.clonedeep'

import WalletProvider from '@openworklabs/filecoin-wallet-provider'
import { FilecoinNumber } from '@openworklabs/filecoin-number'
import { Provider } from 'react-redux'

import { NetworkCheck } from '../lib/check-network'
import BalancePoller from '../lib/update-balance'
import { ConverterContext } from '../lib/Converter'
import { theme, ThemeProvider } from '../components/Shared'
import { initializeStore } from './index'
import { initialState } from '../store/states'
import walletProviderReducer, {
  initialState as walletProviderInitialState,
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../WalletProvider/state'
import { WalletProviderContext } from '../WalletProvider'
import createPath from '../utils/createPath'
import { IMPORT_MNEMONIC } from '../constants'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const mockRouter = jest.fn(() => {})
useRouter.mockImplementation(() => ({
  query: 'network=t',
  push: mockRouter,
  replace: mockRouter
}))

jest.mock('@openworklabs/filecoin-wallet-provider')
const mockGetAccounts = jest
  .fn()
  .mockImplementation(() =>
    Promise.resolve(['t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'])
  )

const mockGetBalance = jest
  .fn()
  .mockImplementation(() => Promise.resolve(new FilecoinNumber('1', 'fil')))

WalletProvider.mockImplementation(() => {
  return {
    wallet: {
      getAccounts: mockGetAccounts,
      sign: jest.fn().mockImplementation(() => 'xxxyyyyzzzz')
    },
    getBalance: mockGetBalance,
    getNonce: jest.fn().mockImplementation(() => 0),
    estimateGas: jest
      .fn()
      .mockImplementation(() => new FilecoinNumber('122', 'attofil')),
    sendMessage: jest.fn().mockImplementation(() => 'QmZCid!')
  }
})

const mockWalletProviderInstance = new WalletProvider()

const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  })
}

const composeWalletProviderState = (initialWalletProviderState, preset) => {
  switch (preset) {
    case 'postOnboard': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    default:
      return initialWalletProviderState
  }
}

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const mockConverterInstance = options.mockConverterInstance || {
    cacheConversionRate: jest.fn(),
    toFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').dividedBy(5)
    }),
    fromFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').multipliedBy(5)
    })
  }

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )
    return (
      <Provider store={store}>
        <ConverterContext.Provider
          value={{
            converter: mockConverterInstance,
            converterError: options.converterError || null
          }}
        >
          <WalletProviderContext.Provider
            value={{
              state: walletProviderState,
              dispatch: walletProviderDispatch,
              fetchDefaultWallet: jest.fn().mockImplementation(),
              setWalletError: jest
                .fn()
                .mockImplementation(errorMessage =>
                  walletProviderDispatch(setError(errorMessage))
                ),
              setWalletType: jest
                .fn()
                .mockImplementation(walletType =>
                  walletProviderDispatch(setWalletType(walletType))
                ),
              setLedgerProvider: jest.fn(),
              connectLedger: jest.fn(),
              resetLedgerState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetLedgerState())
              }),
              resetState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetState())
              })
            }}
          >
            <NetworkCheck
              networkFromRdx={store.getState().network}
              pathname={pathname}
              query={query}
              switchNetwork={jest.fn()}
            />
            <BalancePoller />
            <ThemeProvider theme={theme}>{children}</ThemeProvider>
          </WalletProviderContext.Provider>
        </ConverterContext.Provider>
      </Provider>
    )
  }

  return { Tree, store, walletProvider: mockWalletProviderInstance }
}
