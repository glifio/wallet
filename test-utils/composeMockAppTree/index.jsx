/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import { Provider } from 'react-redux'
import { Converter } from '@openworklabs/filecoin-number'

import { NetworkCheck } from '../../lib/check-network'
import BalancePoller from '../../lib/update-balance'
import { ConverterContext } from '../../lib/Converter'
import { theme, ThemeProvider } from '../../components/Shared'
import { initializeStore } from '..'
import walletProviderReducer, {
  initialState as walletProviderInitialState
} from '../../WalletProvider/state'
import { WasmContext } from '../../lib/WasmLoader'
import { WalletProviderContext } from '../../WalletProvider'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'
import createMockWalletProviderContextFuncs from './createWalletProviderContextFuncs.js'
import mockWalletSubproviders from '../mocks/mock-wallet-subproviders'
import { presets, composeWalletProviderState } from './composeState'

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * walletProviderDispatch(a dispatcher to mock/read calls to the walletProvider's state)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )

    const mockWalletProviderContextFuncs = createMockWalletProviderContextFuncs(
      options.walletProviderDispatch || walletProviderDispatch
    )

    return (
      <Provider store={store}>
        <WasmContext.Provider
          value={{
            generateMnemonic: jest
              .fn()
              .mockImplementation(
                () =>
                  'slender spread awkward chicken noise useful thank dentist tip bronze ritual explain version spot collect whisper glow peanut bus local country album punch frown'
              )
          }}
        >
          <ConverterContext.Provider
            value={{
              converter: new Converter(),
              converterError: options.converterError || null
            }}
          >
            <WalletProviderContext.Provider
              value={{
                state: walletProviderState,
                dispatch:
                  options.walletProviderDispatch || walletProviderDispatch,
                walletSubproviders: {
                  ...mockWalletSubproviders
                },
                ...mockWalletProviderContextFuncs
              }}
            >
              <NetworkCheck
                networkFromRdx={store.getState().network}
                pathname={pathname}
                query={query}
                switchNetwork={jest.fn()}
              />
              <BalancePoller />
              <ThemeProvider theme={theme}>{children}</ThemeProvider>
            </WalletProviderContext.Provider>
          </ConverterContext.Provider>
        </WasmContext.Provider>
      </Provider>
    )
  }

  return {
    Tree,
    store,
    walletProvider: mockWalletProviderInstance
  }
}
