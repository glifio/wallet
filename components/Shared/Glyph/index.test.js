import { cleanup, render, act, fireEvent } from '@testing-library/react'
import composeMockAppTree from '../../../test-utils/composeMockAppTree'
import Glyph from '.'
import theme from '../theme'

describe('Glyph', () => {
  afterEach(cleanup)
  test('renders the glyph', () => {
    const { Tree } = composeMockAppTree('preOnboard')
    const { container } = render(
      <Tree>
        <Glyph
          acronym='Sw'
          bg='core.primary'
          borderColor='core.primary'
          color='core.white'
        />
      </Tree>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the glyph with the text', () => {
    const { Tree } = composeMockAppTree('preOnboard')
    const { getByText } = render(
      <Tree>
        <Glyph
          acronym='Sw'
          bg='core.primary'
          borderColor='core.primary'
          color='core.white'
        />
      </Tree>
    )

    expect(getByText('Sw')).toBeTruthy()
  })

  test('renders a Glyph with correct colors', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const { container } = render(
      <Tree>
        <Glyph acronym='Sw' bg='core.secondary' color='core.white' />
      </Tree>
    )

    const div = container.querySelector('div')
    expect(div).toHaveStyle(
      `color: ${theme.colors.core.white};
      background-color: ${theme.colors.core.secondary};`
    )
  })
})
