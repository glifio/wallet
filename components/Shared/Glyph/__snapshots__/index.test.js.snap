// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glyph renders the glyph 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-radius: 8px;
  border-style: solid;
  border-color: #5E26FF;
  background-color: #5E26FF;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

<div
  class="c0"
  color="core.white"
  display="flex"
  size="6"
>
  <h3
    class="c1"
  >
    Sw
  </h3>
</div>
`;
